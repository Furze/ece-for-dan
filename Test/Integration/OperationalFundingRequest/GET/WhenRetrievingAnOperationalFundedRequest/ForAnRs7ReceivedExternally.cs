using System;
using System.Collections.Generic;
using Bard;
using MoE.ECE.Domain.Read.Model.OperationalFunding;
using MoE.ECE.Integration.Tests.Chapter;
using MoE.ECE.Integration.Tests.Infrastructure;
using Shouldly;
using Xunit;
using Xunit.Abstractions;

namespace MoE.ECE.Integration.Tests.OperationalFundingRequest.GET.WhenRetrievingAnOperationalFundedRequest
{
    public class ForAnRs7ReceivedExternally : IntegrationTestBase<ECEStoryBook, ECEStoryData>
    {
        public ForAnRs7ReceivedExternally(RunOnceBeforeAllTests testSetUp, ITestOutputHelper output,
            TestState<ECEStoryBook, ECEStoryData> testState) : base(testSetUp, output, testState)
        {
        }

        private Guid _businessEntityId;

        protected override void Arrange() =>
            Given
                .An_rs7_has_been_received_externally()
                .GetResult(data => _businessEntityId = data.Rs7Model.BusinessEntityId.GetValueOrDefault());

        protected override void Act() =>
            When.Get(
                $"api/operational-funding-requests?business-entity-id={_businessEntityId}");

        [Fact]
        public void ThenTheResponseShouldBeOk() => Then.Response.ShouldBe.Ok();

        [Fact]
        public void ThenTheResponseShouldContain1Item()
        {
            var result = Then.Response.Content<ICollection<OperationalFundingRequestModel>>();

            result.Count.ShouldBe(1);
        }
        
        [Fact]
        public void Then_the_response_snapshot_should_be_ok() =>
            Then.Snapshot().Match<ICollection<OperationalFundingRequestModel>>(IgnoreFieldsFor.CollectionOperationalFundingRequestModel);
    }
}