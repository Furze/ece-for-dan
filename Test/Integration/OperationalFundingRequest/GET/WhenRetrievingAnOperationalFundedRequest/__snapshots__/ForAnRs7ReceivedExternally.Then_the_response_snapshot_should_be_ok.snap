[
  {
    "EntitlementMonths": [
      {
        "MonthName": "February 2020",
        "MonthNumber": 2,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 0,
        "AllDayNonCertificatedTeacherHours": 0,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 0,
        "WashUpPlusTen": -2800.8,
        "WashUpUnderTwo": -9343.35,
        "WashUpTwoAndOver": -4959.75,
        "WashUpTwentyHours": -7750.05,
        "TotalEntitlement": 0.0,
        "TotalFundsAdvanced": 24853.95,
        "TotalEntitlementUnderTwo": 0.0,
        "TotalEntitlementTwoAndOver": 0.0,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 0.0,
        "TotalWashUp": -24853.95,
        "FundingComponents": []
      },
      {
        "MonthName": "March 2020",
        "MonthNumber": 3,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 0,
        "AllDayNonCertificatedTeacherHours": 0,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 0,
        "WashUpPlusTen": 0.0,
        "WashUpUnderTwo": 0.0,
        "WashUpTwoAndOver": 0.0,
        "WashUpTwentyHours": 0.0,
        "TotalEntitlement": 0.0,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": 0.0,
        "TotalEntitlementTwoAndOver": 0.0,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": 0.0,
        "TotalWashUp": 0.0,
        "FundingComponents": []
      },
      {
        "MonthName": "April 2020",
        "MonthNumber": 4,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 5,
        "AllDayNonCertificatedTeacherHours": 6,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 1,
        "WashUpPlusTen": 19.04,
        "WashUpUnderTwo": 45.8,
        "WashUpTwoAndOver": 14.28,
        "WashUpTwentyHours": 18.94,
        "TotalEntitlement": 98.06,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": 45.8,
        "TotalEntitlementTwoAndOver": 14.28,
        "TotalEntitlementTwentyHours": 18.94,
        "TotalEntitlementPlusTen": 19.04,
        "TotalWashUp": 98.06,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 45.8,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 5,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 14.28,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 3,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 18.94,
            "Rate": 9.47,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-04-01T00:00:00",
            "Amount": 19.04,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 4,
            "OperatingDays": 1
          }
        ]
      },
      {
        "MonthName": "May 2020",
        "MonthNumber": 5,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 5,
        "AllDayNonCertificatedTeacherHours": 6,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 1,
        "WashUpPlusTen": 19.04,
        "WashUpUnderTwo": 45.8,
        "WashUpTwoAndOver": 14.28,
        "WashUpTwentyHours": 18.94,
        "TotalEntitlement": 98.06,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": 45.8,
        "TotalEntitlementTwoAndOver": 14.28,
        "TotalEntitlementTwentyHours": 18.94,
        "TotalEntitlementPlusTen": 19.04,
        "TotalWashUp": 98.06,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 45.8,
            "Rate": 9.16,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 5,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 14.28,
            "Rate": 4.76,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 3,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 18.94,
            "Rate": 9.47,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-05-01T00:00:00",
            "Amount": 19.04,
            "Rate": 4.76,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 4,
            "OperatingDays": 1
          }
        ]
      }
    ],
    "OrganisationId": 114895,
    "BusinessEntityId": "a420324a-2ace-4b2c-9e15-c7dcbf88c104",
    "RequestId": "6131",
    "FundingPeriodMonth": "July",
    "FundingYear": 2021,
    "RevisionNumber": 1,
    "TotalWashUp": -24657.83,
    "TotalAdvance": 632.03,
    "AdvanceMonths": [
      {
        "MonthName": "July 2020",
        "MonthNumber": 7,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 35.06,
        "AmountPayableUnderTwo": 84.33,
        "AmountPayableTwoAndOver": 26.3,
        "AmountPayableTwentyHours": 34.88,
        "TotalDays": 2,
        "TotalAdvance": 180.57,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-07-01T00:00:00",
            "Amount": 84.33,
            "Rate": 9.37,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 10,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-07-01T00:00:00",
            "Amount": 26.3,
            "Rate": 4.87,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 6,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-07-01T00:00:00",
            "Amount": 34.88,
            "Rate": 9.69,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 4,
            "OperatingDays": 2
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-07-01T00:00:00",
            "Amount": 35.06,
            "Rate": 4.87,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 8,
            "OperatingDays": 2
          }
        ]
      },
      {
        "MonthName": "August 2020",
        "MonthNumber": 8,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 1,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 17.53,
        "AmountPayableUnderTwo": 42.17,
        "AmountPayableTwoAndOver": 13.15,
        "AmountPayableTwentyHours": 17.44,
        "TotalDays": 1,
        "TotalAdvance": 90.29,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-08-03T00:00:00",
            "Amount": 42.17,
            "Rate": 9.37,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 5,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-08-03T00:00:00",
            "Amount": 13.15,
            "Rate": 4.87,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 3,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-08-03T00:00:00",
            "Amount": 17.44,
            "Rate": 9.69,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-08-03T00:00:00",
            "Amount": 17.53,
            "Rate": 4.87,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 4,
            "OperatingDays": 1
          }
        ]
      },
      {
        "MonthName": "September 2020",
        "MonthNumber": 9,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 3,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 52.6,
        "AmountPayableUnderTwo": 126.5,
        "AmountPayableTwoAndOver": 39.45,
        "AmountPayableTwentyHours": 52.33,
        "TotalDays": 3,
        "TotalAdvance": 270.88,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-09-01T00:00:00",
            "Amount": 126.5,
            "Rate": 9.37,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 15,
            "OperatingDays": 3
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-09-01T00:00:00",
            "Amount": 39.45,
            "Rate": 4.87,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 9,
            "OperatingDays": 3
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-09-01T00:00:00",
            "Amount": 52.33,
            "Rate": 9.69,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 6,
            "OperatingDays": 3
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-09-01T00:00:00",
            "Amount": 52.6,
            "Rate": 4.87,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 12,
            "OperatingDays": 3
          }
        ]
      },
      {
        "MonthName": "October 2020",
        "MonthNumber": 10,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 1,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 17.53,
        "AmountPayableUnderTwo": 42.17,
        "AmountPayableTwoAndOver": 13.15,
        "AmountPayableTwentyHours": 17.44,
        "TotalDays": 1,
        "TotalAdvance": 90.29,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "2020-10-01T00:00:00",
            "Amount": 42.17,
            "Rate": 9.37,
            "RateName": "Under2ADTLcentrebased25-49",
            "FundedChildHours": 5,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "2020-10-01T00:00:00",
            "Amount": 13.15,
            "Rate": 4.87,
            "RateName": "2AndOverADTLcentrebased25-49",
            "FundedChildHours": 3,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "2020-10-01T00:00:00",
            "Amount": 17.44,
            "Rate": 9.69,
            "RateName": "20HoursADTLcentrebased25-49",
            "FundedChildHours": 2,
            "OperatingDays": 1
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "2020-10-01T00:00:00",
            "Amount": 17.53,
            "Rate": 4.87,
            "RateName": "Plus10ADTLcentrebased25-49",
            "FundedChildHours": 4,
            "OperatingDays": 1
          }
        ]
      }
    ],
    "MatchingAdvanceMonths": [
      {
        "MonthName": "February",
        "MonthNumber": 2,
        "Year": 2020,
        "ParentLedWorkingDays": null,
        "AllDayWorkingDays": 20,
        "SessionalWorkingDays": 0,
        "AmountPayablePlusTen": 2800.8,
        "AmountPayableUnderTwo": 9343.35,
        "AmountPayableTwoAndOver": 4959.75,
        "AmountPayableTwentyHours": 7750.05,
        "TotalDays": null,
        "TotalAdvance": null,
        "FundingComponents": [
          {
            "SessionType": "AllDay",
            "FundingComponentType": "PlusTen",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 2800.8,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "UnderTwo",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 9343.35,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwoAndOver",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 4959.75,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          },
          {
            "SessionType": "AllDay",
            "FundingComponentType": "TwentyHours",
            "StartDate": "0001-01-01T00:00:00",
            "Amount": 7750.05,
            "Rate": 10.1,
            "RateName": "",
            "FundedChildHours": 30,
            "OperatingDays": 20
          }
        ]
      }
    ],
    "Equity": null,
    "BusinessExceptions": []
  }
]
